src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -294,6 +306,38 @@ impl TargetInfo {
                 gctx.shell().warn("non-trivial mutual dependency between target-specific configuration and RUSTFLAGS")?;
             }
 
+            let mut supports_std: Option<bool> = None;
+
+            // The '--print=target-spec-json' is an unstable option of rustc, therefore if
+            // rustc is not a 'Prerelease' build, it's safe to assume that we cannot fetch
+            // the target-spec-json. So don't bother trying.
+            if rustc.version.is_prerelease() {
```

Comment:
We normally use [`GlobalContext::nightly_features_allowed`](https://doc.rust-lang.org/nightly/nightly-rustc/cargo/util/context/struct.GlobalContext.html#structfield.nightly_features_allowed) to detect it. Any reason preferring `is_prerelease` over that?

Reply:
No reason, I wasn't aware of the preferred method for detecting this. I'll make this change.